openedx/features/course_experience/views/welcome_message.py

```diff
@@ -28,7 +28,7 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         Returns: A fragment, or None if there is no welcome message.
         """
         course_key = CourseKey.from_string(course_id)
-        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
```

Comment:
This was failing because the fragment was being rendered but then not included in the page. Now that the course home fragment doesn't render fragments that it doesn't need, this change isn't needed. However, I find it odd that the fragment would be verifying access and issuing page redirects. It would be better if the fragment itself just didn't render itself, but I don't want to make that change in this PR.